### PR TITLE
Interpolate

### PR DESCRIPTION
**Pull request by EthanNichols**
Increase the rate which packets are sent from the server to clients to 0.1
Also increase the rate at which client inputs are sent to the server to 0.1

Added Interpolation struct to Entity.
This struct should be moved/changed to only apply to network replicated entities.

### CODE REVIEW

File: Weave_Common/Entity/Entity.h [hunk 1]
```diff
@@ -117,6 +119,67 @@ class Entity : public IPoolable< Entity >, public ISaveable
...
+		/// Starting and ending position to lerp between
+		glm::vec3 finalPos = glm::vec3(0, 0, 0);
+		glm::vec3 startPos = glm::vec3(0, 0, 0);
+
+		/// Starting and ending rotations to lerp between
+		glm::vec3 finalRot = glm::vec3(0, 0, 0);
+		glm::vec3 startRot = glm::vec3(0, 0, 0);
+
+		/// Lerp times
+		float packetTime = 0;
```
Comment (BenjaFriend) [inline Weave_Common/Entity/Entity.h]: coercion

File: Weave_Common/Entity/Entity.h [hunk 2]
```diff
@@ -117,6 +119,67 @@ class Entity : public IPoolable< Entity >, public ISaveable
 
 protected:
 
+	/// Ideally this would be moved to a networking transform/component
+	/// Since client side entities don't necessarily need to have
+	/// interpolation applied to changes with the transform
+	struct Interpolation
+	{
+	public:
```
Comment (BenjaFriend) [inline Weave_Common/Entity/Entity.h]: Why no constructor for this?

File: Weave_Engine/TankGame/Game/PlayerMoves.cpp [hunk 3]
```diff
@@ -30,4 +31,35 @@ PlayerMoves::PlayerMoves()
...
 {
 
+}
+
+void PlayerMoves::QueueCurrentMoves()
+{
+	for (auto& move : currentMoves)
+	{
+		MoveQueue.push_back(move);
+		move.time = 0;
```
Comment (BenjaFriend) [inline Weave_Engine/TankGame/Game/PlayerMoves.cpp]: coercion

File: Weave_Engine/TankGame/Game/PlayerMoves.h [hunk 4]
```diff
@@ -27,22 +33,31 @@ namespace Tanks
...
+		void Update(float deltaTime);
 
     private:
 
+		static Input::InputManager* inMan;
+
         // Keep track of what buttons have been pressed
-        std::deque<Input::InputType> MoveQueue;
+        std::deque<TimedMove> MoveQueue;
+		std::vector<TimedMove> currentMoves;
```
Comment (BenjaFriend) [inline Weave_Engine/TankGame/Game/PlayerMoves.h]: Formatting issues

File: Weave_Engine/Weave_Engine/Scenes/Scene.h [hunk 5]
```diff
@@ -35,6 +35,8 @@ namespace SceneManagement
 
         virtual void Read( InputMemoryBitStream& inInputStream ) override;
 
+		float packetTripTime = 0;
```
Comment (BenjaFriend) [inline Weave_Engine/Weave_Engine/Scenes/Scene.h]: coercion

File: Weave_Server/Weave_Server/src/ServerNetworkManager.cpp [hunk 6]
```diff
@@ -143,7 +143,9 @@ void ServerNetworkManager::ProcessInputPacket(ClientProxyPtr aClient, InputMemor
     for ( size_t i = 0; i < sizeOfMoveList; ++i )
     {
         UINT8 move = 0;
+		float time = 0;
```
Comment (BenjaFriend) [inline Weave_Server/Weave_Server/src/ServerNetworkManager.cpp]: coercion

File: Weave_Common/Entity/Entity.h [hunk 7]
```diff
@@ -29,6 +29,8 @@ class Entity : public IPoolable< Entity >, public ISaveable
         EIRS_AllState = EIRS_POS | EIRS_ROT
     };
 
+	float packetTripTime = 0;
```
Comment (BenjaFriend) [inline Weave_Common/Entity/Entity.h]: coercion

**Review by BenjaFriend — CHANGES_REQUESTED** [review]
There are a couple of things going on that I think should be changed.
* There is a lot of coercion (implicit type conversion) with your float values
* The formatting looks weird on here (the tabs and things are all offset?)
* I would like to go in and just format things so that the codebase is consistent in its style

**Comment by EthanNichols** [comment]
Should just require formatting. Since I don't have the right setup, you can do that.

**Pull request merged**